integration-tests/jpa/src/main/java/org/apache/camel/quarkus/component/jpa/it/JpaRoute.java

```diff
@@ -73,7 +77,12 @@ public void configure() throws Exception {
         from("direct:idempotent")
                 .idempotentConsumer(
                         header("messageId"),
-                        jpaMessageIdRepository(entityManagerFactory, "idempotentProcessor"))
+                        new JpaMessageIdRepository(entityManagerFactory, new TransactionStrategy() {
+                            @Override
+                            public void executeInTransaction(Runnable runnable) {
+                                QuarkusTransaction.run(runOptions().semantic(RunOptions.Semantic.JOIN_EXISTING), runnable);
+                            }
+                        }, "idempotentProcessor"))
```

Comment:
Will end users always have to create the `JpaMessageIdRepository` like this? If so, can't we automate it for them somehow? Or at least document that they need to do this?

Reply:
yeah, I think we can add `QuarkusTransactionStrategy`.

Reply:
Done